### PR TITLE
LibCore+Everywhere: Remove ArgsParser::add*(char const*&)

### PR DESCRIPTION
**Pull request by alimpfard**
This is not guaranteed to always work correctly as ArgsParser deals in StringViews and might have a non-properly-null-terminated string as a value. As a bonus, using StringView (and DeprecatedString where necessary) leads to nicer looking code too :^)

### CODE REVIEW

File: Userland/Utilities/xargs.cpp [hunk 1]
```diff
@@ -44,10 +44,10 @@ ErrorOr<int> serenity_main(Main::Arguments main_arguments)
 
     StringView placeholder;
     bool split_with_nulls = false;
-    char const* specified_delimiter = "\n";
+    DeprecatedString specified_delimiter = "\n"sv;
     Vector<DeprecatedString> arguments;
     bool verbose = false;
-    char const* file_to_read = "-";
+    DeprecatedString file_to_read = "-"sv;
```
Comment (kleinesfilmroellchen) [inline Userland/Utilities/xargs.cpp:50]: Why are these not StringViews?

**Review by kleinesfilmroellchen — CHANGES_REQUESTED** [review]
The master of all Serenity C haters approves of this :^)

File: Userland/Utilities/xargs.cpp (hunk 1 above)
Comment (alimpfard) [reply Userland/Utilities/xargs.cpp:50]: Because they're used later on in C APIs.

Comment (kleinesfilmroellchen) [reply Userland/Utilities/xargs.cpp:50]: huh I thought i checked, consider this comment null and void

File: Userland/Utilities/syscall.cpp [hunk 2]
```diff
@@ -155,36 +155,34 @@ static FlatPtr parse_parameter_buffer(ArgIter& iter)
...
     }
 
     // Is it the output buffer?
-    if (strcmp(this_arg, "buf") == 0)
+    if (this_arg_string == "buf"sv)
         return (FlatPtr)outbuf;
 
     // Is it a parameter buffer?
-    if (strcmp(this_arg, "[") == 0)
+    if (this_arg == "["sv)
```
Comment (AtkinsSJ) [inline Userland/Utilities/syscall.cpp:174]: I think we want to compare `this_arg_string` here?

**Review by AtkinsSJ — APPROVED** [review]

**Pull request merged**